sky/packages/sky/lib/src/widgets/navigator.dart

```diff
@@ -52,6 +52,7 @@ class NavigatorState extends State<Navigator> {
     assert(route != null);
     assert(!route.ephemeral);
     _history.add(route);
```

Comment:
Remove the _history.add(route), and instead do the _history.insert inside _didPushRoute, so you can have the "did push" logic in the same function as the "actually push" function, to avoid the risk of forgetting to call it some time.